globus_cli/commands/ls.py

```diff
@@ -38,9 +60,15 @@ def ls_command(endpoint_plus_path, recursive_depth_limit,
     autoactivate(client, endpoint_id, if_expires_in=60)
 
     # create the query paramaters to send to operation_ls
-    ls_params = {"show_hidden": int(all)}
+    ls_params = {"show_hidden": int(show_hidden)}
     if path:
         ls_params["path"] = path
+    if filter_val:
+        # these chars have special meaning in the LS API's filter clause
+        if ':' in filter_val or '/' in filter_val:
```

Comment:
Transfer does `pair.split(':', 1)` on the string after splitting on the slash, so I don't *think* you *have to* forbid the colon character.

I tried commenting out this safety check, throwing a `fi:le3.txt` into my home directory, and ran `globus ls ddb59aef-6d04-11e5-ba46-22000b92c6ec --filter='fi:le3.txt'` and `globus ls ddb59aef-6d04-11e5-ba46-22000b92c6ec --filter='~*:*'`; those worked fine.

The slash would still need to be forbidden, but such a filename isn't generally supported anyway.

Reply:
I had considered, but not dared, to try this. I think it's a good change and will make it.